src/debug.c

```diff
@@ -2305,16 +2409,19 @@ void serverLogHexDump(int level, char *descr, void *value, size_t len) {
 /* =========================== Software Watchdog ============================ */
 #include <sys/time.h>
 
-void watchdogSignalHandler(int sig, siginfo_t *info, void *secret) {
+void sigalrmSignalHandler(int sig, siginfo_t *info, void *secret) {
 #ifdef HAVE_BACKTRACE
     ucontext_t *uc = (ucontext_t*) secret;
 #else
     (void)secret;
 #endif
-    UNUSED(info);
     UNUSED(sig);
 
-    serverLogFromHandler(LL_WARNING,"\n--- WATCHDOG TIMER EXPIRED ---");
+    /* SIGALRM can be sent explicitly to the process calling kill() to get the stacktraces,
+    or every watchdog_period interval. In the last case, si_pid is not set */
+    if(info->si_pid == 0) {
+        serverLogFromHandler(LL_WARNING,"\n--- WATCHDOG TIMER EXPIRED ---");
```

Comment:
maybe we should have an `else` that prints some other title explaining why there's a stacktrace on the log (i.e. even some "SIGALRM received".
also, maybe print `si_code`, `si_pid` etc, like we do on SIGSEGV?

Reply:
I added "Received SIGALR" to avoid using unsafe formated string function 

>  * We actually use this only for signals that are not fatal from the point
 * of view of Redis. Signals that are going to kill the server anyway and
 * where we need printf-alike features are served by serverLog(). */
void serverLogFromHandler(int level, const char *msg) {